hooks/tk-multi-publish2/ingest/collector.py

```diff
@@ -606,11 +622,88 @@ def _collect_manifest_file(self, settings, parent_item, path):
 
         return file_items
 
+    def on_context_changed(self, settings, item):
+        """
+        Callback to update the item on context changes.
+
+        :param dict settings: Configured settings for this collector
+        :param item: The Item instance
+        """
+
+        # get the item fields and everything in place
+        super(IngestCollectorPlugin, self).on_context_changed(settings, item)
+
+        # restore the fields from the manifest file, even though we currently don't allow users to change context
+        # when ingesting using the manifest file.
+        if "manifest_file_fields" in item.properties:
```

Comment:
This should all be moved to a `_resolve_item_fields()` method override.

Reply:
Yep, will move this to `_resolve_item_fields()`.